compiler/shared_ast/typing.ml

```diff
@@ -959,17 +957,10 @@ and typecheck_expr_top_down :
     in
     Expr.ecustom obj targs tret mark
 
-let wrap ctx f e =
-  try f e
-  with Type_error (e, ty1, ty2) -> (
-    let bt = Printexc.get_raw_backtrace () in
-    try handle_type_error ctx e ty1 ty2
-    with e -> Printexc.raise_with_backtrace e bt)
-
-let wrap_expr ctx f e =
+let wrap_expr f e =
   (* We need to unbox here, because the typing may otherwise be stored in
      Bindlib closures and not yet applied, and would escape the `try..with` *)
```

Comment:
Comment about `try..with` no longer appropriate ; probably the `unbox` can be inlined at the call sites.

Reply:
Fixed